src/vs/platform/files/test/electron-browser/diskFileService.test.ts

```diff
@@ -1674,6 +1674,13 @@ flakySuite('Disk File Service', function () {
 		assert.strictEqual(event!.target!.resource.fsPath, resource.fsPath);
 	});
 
+	(isWindows ? test : test.skip)('createFile (Windows does not allow files to end with ".")', async () => {
+		const resource = URI.file(join(testDir, 'test.'));
+		await assert.rejects(async () => {
+			await service.writeFile(resource, VSBuffer.fromString(''));
+		});
+	});
+
```

Comment:
I didn't see any other test in this file using `assert.rejects`, but I think it is the correct way to go.  If not, I can use another method.